util/util_tls.go

```diff
@@ -0,0 +1,171 @@
+			"certKey":  certPath,
+		}).Debug("skipping self-cert creation, files exist")
+		return nil
+	}
+
+	certRoot := filepath.Dir(certPath)
+	keyRoot := filepath.Dir(keyPath)
+	if err := os.MkdirAll(certRoot, 0755); err != nil {
+		ctx.WithFields(log.Fields{
+			"host":     host,
+			"certRoot": certRoot,
+		}).Debug("created dir")
+
+		return err
+	}
+	if keyRoot != certRoot {
+		if err := os.MkdirAll(keyRoot, 0755); err != nil {
+			ctx.WithFields(log.Fields{
+				"host":    host,
+				"keyRoot": keyRoot,
+			}).Debug("created dir")
+			return err
+		}
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return err
+	}
+
+	tmpl := x509.Certificate{
+		SerialNumber:       serialNumber,
+		Subject:            pkix.Name{Organization: []string{orgName}, CommonName: host},
+		NotBefore:          time.Now(),
+		NotAfter:           time.Now().AddDate(1, 0, 0),
+		SignatureAlgorithm: x509.SHA256WithRSA,
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	if ip := net.ParseIP(host); ip != nil {
+		ctx.WithField("ip", ip).Debug("ip used in cert template")
+		tmpl.IPAddresses = append(tmpl.IPAddresses, ip)
+	} else {
+		ctx.WithField("host", host).Debug("host used in cert template")
+		tmpl.DNSNames = append(tmpl.DNSNames, host)
+	}
+
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	// gen cert file
+	ctx.WithField("certFile", certPath).Debug("creating cert file")
+	certBlock, err := x509.CreateCertificate(rand.Reader, &tmpl, &tmpl, &privKey.PublicKey, privKey)
+	if err != nil {
+		return err
+	}
+
+	certFile, err := os.Create(certPath)
+	if err != nil {
+		return err
+	}
+	defer certFile.Close()
+	if err := pem.Encode(certFile, &pem.Block{Type: certBlockType, Bytes: certBlock}); err != nil {
+		return err
+	}
+
+	// gen key file
+	ctx.WithField("keyFile", keyPath).Debug("creating key file")
+	keyFile, err := os.OpenFile(keyPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+	defer keyFile.Close()
+
+	keyBlock := x509.MarshalPKCS1PrivateKey(privKey)
+	if err != nil {
+		return err
+	}
+
+	if err := pem.Encode(keyFile, &pem.Block{Type: keyBlockType, Bytes: keyBlock}); err != nil {
+		return err
+	}
+
+	ctx.WithFields(log.Fields{
+		"certPath": certPath,
+		"certKey":  certPath,
+	}).Debug("self-cert files created")
+
+	return nil
+}
+
+func AssertTrustedHost(ctx apitypes.Context, host, algo string, fingerprint []byte) bool {
+	trusted := "no"
+	fmt.Printf("Rejecting connection to unknown host %s.\n", host)
+	fmt.Printf("SHA Fingerprint presented: %s:%x.\n", algo, fingerprint[0:32])
+	fmt.Print("Do you want to save host to knwon_hosts file? (yes/no): ")
+	fmt.Scanf("%s", &trusted)
+	if strings.ToLower(trusted) == "yes" {
```

Comment:
Hi @vladimirvivien,

A couple of comments here:

1. Please refrain from ever using `strings.To(Lower|Upper) ==` to compare string values. Here is an [old](https://github.com/codedellemc/rexray/pull/540#r76169894) comment I made to Proud about the topic:

> I recently read [an article](https://medium.com/@benbjohnson/go-walkthrough-bytes-strings-packages-499be9f4b5bd) by @benbjohnson on ways to optimize Go code, and string comparisons are one area that can be greatly improved. To achieve efficient, case-insensitive string comparison, please use [`string.EqualFold`](https://golang.org/pkg/strings/#EqualFold) instead. Memory is allocated with every `strings.ToLower` call, and this is avoided by using the `EqualFold` comparison function. I've started to use it wherever I would normally have used a `ToLower` or `ToUpper` to perform case-insensitive comparison. 

2. Please ensure that variations of `yes` are valid answers as well. My suggestion is to use a regexp like this example from the Golang playground ([run](https://play.golang.org/p/pOGdvRRJyU)). Essentially something like this: ``if ok, _ := regexp.MatchString(`(?i)^y(?:es)?$`, trusted); ok {`` It's case insensitive and accepts just `y|Y` or any cased version of the word `yes`.